module/canvas/detection-modes/see-invisibility.mjs

```diff
@@ -0,0 +1,158 @@
+/**
+ * The detection mode for See Invisibility.
+ */
+export class DetectionModeSeeInvisibility extends DetectionMode {
+  constructor() {
+    super({
+      id: "seeInvisibility",
+      label: "DETECTION.SeeInvisibility",
+      type: DetectionMode.DETECTION_TYPES.SIGHT,
+      walls: false
+    });
+  }
+
+  /** @override */
+  static getDetectionFilter() {
+    return this._detectionFilter ??= GlowOverlayFilter.create({
+      glowColor: [0, 0.60, 0.33, 1]
+    });
+  }
+
+  /** @override */
+  _canDetect(visionSource, target) {
+    // Only invisible tokens can be detected.
+    return target instanceof Token && target.document.hasStatusEffect(CONFIG.specialStatusEffects.INVISIBLE);
+  }
+
+  /** @override */
+  _testPoint(visionSource, mode, target, test) {
+    if (!super._testPoint(visionSource, mode, target, test)) {
+      return false;
+    }
+
+    const visionSources = this.#removeOtherVisionSources(visionSource);
+    const detectionsModes = this.#removeNonSightDetectionModes(visionSource);
+    const activeEffects = this.#removeInvisibleStatusEffects(target);
+
+    // Test whether this vision source sees the target without the invisible status effect.
+    const isVisible = canvas.effects.visibility.testVisibility(test.point, { tolerance: 0, object: target });
+
+    this.#restoreOtherVisionSources(visionSources);
+    this.#restoreNonSightDetectionModes(visionSource, detectionsModes);
+    this.#restoreInvisibleStatusEffects(target, activeEffects);
```

Comment:
Rather than removing and restoring sources/detection modes/effects, would it be possible to instead extend or modify `testVisibility` so that these things can be ignored in its logic?

Reply:
Vision source and detection mode filtering could be done in `testVisibility` without removing and restoring, but suppressing the status effect cannot. So it wouldn't really be a big improvement. Also there's the issue that having the system override `testVisibility` could cause module compatibility issues with those that override it too.

Reply:
Why does the status effect need to be suppressed?

Reply:
We need the detection modes to be able to detect invisible tokens when they normally cannot. If we don't suppress the invisible status effect, See Invisibility doesn't do anything.